server/routes/notification.js

```diff
@@ -0,0 +1,90 @@
+const authMiddleware = require("../middlewares/authMiddleware");
+const ListProductModel = require("../models/listProductModel");
+const NotificationModel = require("../models/notificationModel");
+const ProductModel = require("../models/productModel");
+const UserListModel = require("../models/userListModel");
+const UserNotificationModel = require("../models/userNotificationsModel");
+const router = require("express").Router();
+
+//Make the given notification id read
+router.get("/read/:id", authMiddleware, async (req, res) => {
+  try {
+    const { id } = req.params;
+    const { userId } = req.headers;
+
+    if (!id) return res.status(400).send({ msg: "Must have an id parameter" });
+    if (!userId)
+      return res.status(400).send({ msg: "Must have a userId in headers" });
+
+    const userNotification = await UserNotificationModel.findOne({
+      where: { notification_id: id, user_id: userId },
+    });
+
+    const updatedUserNotification = await userNotification.update({
+      isRead: true,
+    });
+
+    res.send(updatedUserNotification);
+  } catch (error) {
+    res.status(500).send({
+      msg: "Server Error",
+    });
+  }
+});
+
+//Create a notification using a product id given inside the body
+router.post("/createNotification", async (req, res) => {
+  try {
+    const { id } = req.body;
+
+    //Create the notification model using the productId which is also is updated by a service
+    const newNotif = await NotificationModel.create({
+      productId: id,
+    });
+
+    //The plan here is to get the user_id who should get the notification. The said user_id can
+    //be found through ListProductModel => UserListModel.obj.user_id
+
+    //Get all ListProducts that has the same product_id
+    const listProduct = await ListProductModel.findAll({
+      where: { product_id: id },
+    });
+
+    let promiseArr = [];
+
+    //Asynchronous request for finding all the UserListModel using each list_id
+    for (let x = 0; x < listProduct.length; x++) {
+      promiseArr.push(
+        UserListModel.findOne({ where: { id: listProduct[x].list_id } })
+      );
+    }
+
+    const userList = await Promise.all(promiseArr);
+
+    promiseArr = [];
```

Comment:
It would be cleaner to declare `promiseArr` as a constant (`const`), and then create a different array for looping over the `userList`. 
When you do what you did in line 64, the previous array still exists in memory, you're just reassigning the reference to something else. This is not a good idea. 
Look into using [`.map()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) instead of looping with a `for` loop and using `forEach` on the array below... it may give you exactly what you want 😉